src/test/java/com/fasterxml/uuid/JugNoArgsTest.java

```diff
@@ -0,0 +1,217 @@
+package com.fasterxml.uuid;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import java.lang.reflect.Array;
+import java.util.*;
+
+import static org.hamcrest.core.StringContains.containsString;
+import static org.junit.Assert.*;
+
+@RunWith(Parameterized.class)
+public class JugNoArgsTest {
+    @Parameterized.Parameter
+    public String useCase;
+
+    private PrintStream oldStrOut;
+    private PrintStream oldStrErr;
+
+    private final ByteArrayOutputStream outContent = new ByteArrayOutputStream();
+    private final ByteArrayOutputStream errContent = new ByteArrayOutputStream();
+    private Jug jug_underTest;
+
+    @Before
+    public void setup() {
+        jug_underTest = new Jug();
+        oldStrOut = System.out;
+        oldStrErr = System.err;
+        PrintStream stubbedStream = new PrintStream(outContent);
+        System.setOut(stubbedStream);
+        PrintStream stubbedErrStream = new PrintStream(errContent);
+        System.setErr(stubbedErrStream);
+    }
+
+    @After
+    public void cleanup() {
+        System.setOut(oldStrOut);
+        System.setErr(oldStrErr);
+    }
+
+    @Test
+    public void run_givenNoOptions_shouldProduceUUID() {
+        // given
+
+        // when
+        jug_underTest.run(new String[]{useCase});
+
+        // then - if it is a UUID then we should be able to parse it back out
+        String actualUuid = outContent.toString();
+        assertEquals('\n', actualUuid.charAt(actualUuid.length() - 1));
+
+        assertEquals(UUID.class,
+                UUID.fromString(actualUuid.substring(0, actualUuid.length() - 1)).getClass());
```

Comment:
I wasn't super sure how I was going to validate a UUID version, without digging into the actual spec expectation...like what does the first 6 bits mean etc.

But I figured, I would just hook myself into the output stream and as long as I can feed the UUID back into the UUID class then it means its a UUID of some kind...but I am unsure if that makes sense because you actually want to know if the UUID spec is met right? Rather than just checking to see if its UUID-parseable.

Reply:
I don't think we need to thoroughly test validity at this level: these aspects are verified by other, existing unit tests. So "looks about right" is pretty good at CLI level I think. Use of JDK UUID constructor makes sense to me.